### ISSUE TITLE
inspect.getargspec() DeprecationWarning

### ISSUE THREAD

**Issue by miguelmorin** [tensorflow/probability#259]
Running `tensorflow_probability/examples/vae.py` from branch `r0.5`, I get a warning:
```
/venv/lib/python3.6/site-packages/tensorflow/python/util/tf_inspect.py:75: DeprecationWarning: inspect.getargspec() is deprecated, use inspect.signature() or inspect.getfullargspec()
  return _inspect.getargspec(target)
```
The warning returns multiple times during training.

**Comment by csuter** [comment]
Yeah, this is due to usage of deprecated methods upstream in TF core. Unfortunately there's not much we can do about it at the TF Probability level. Might be worth filing an issue on the TF repo, if there isn't one already. It might also be an easy fix, for which I'm sure they'd welcome a PR.

**Comment by SiegeLordEx** [comment]
The "multiple times" aspect of this is likely fixed by https://github.com/tensorflow/probability/commit/70253ba39b62a7bf6d0203a49472b174059556c8.

**Comment by miguelmorin** [comment]
@csuter Thanks for the pointer. I missed that it was in TF core. It's funny because `tf_inspect` does have a wrapper for the newer `getfullargspec`, but other parts of the code still call the old wrapper. After some inspection, I found that the point of either is to avoid keyword and default arguments, and I fixed it by changing lines 135-136 of `tensorflow/python/framework/function.py`from:

```
argspec = tf_inspect.getargspec(func)
if argspec.keywords or argspec.defaults:
```

to 
```
argspec = tf_inspect.getfullargspec(func)
    if argspec.varkw or argspec.defaults or argspec.kwonlyargs or argspec.kwonlydefaults:
```

I will submit a PR to TF Core after I set up the environment with Bazel for the tests.

**Comment by csuter** [comment]
@miguelmorin -- awesome! it would be great to see this fixed in core!

**Comment by wmmxk** [comment]
I also run into this warning and I am using version 1.9

**Comment by miguelmorin** [comment]
Sorry, I was unable to make the TensorFlow tests pass to validate my changes. [This thread](https://github.com/tensorflow/tensorflow/issues/25007#issuecomment-455651448) details my attempts at a build. I got as far as building and then half the tests failed:

```
Executed 1045 out of 1045 tests: 511 tests pass and 534 fail locally.
INFO: Build completed, 534 tests FAILED, 24783 total actions
```

Now I got new projects in the pipeline, Azure is throwing different sorts of errors, and am running out of time. I hope that this will be fixed in TensorFlow 2.
